frontend/apps/core-app/src/clients/coreDBClient.ts

```diff
@@ -0,0 +1,90 @@
+/*
+  USED FOR TESTING PURPOSES ONLY
+*/
+
+import { BaseRESTClient } from 'core-api-client';
```

Comment:
Not sure where to put this comment, but just based on naming (`code-DB-client`), we should probably be dealing with `tables`, `rows`, `fields`, `constraints` rather than `entities`

Reply:
I think the advantage of having a low-level api such as a `DB` api would be to allow clients to interact with lower level concepts that will become necessary for reconciliation/offline use, like row versioning

Reply:
I think the concept of `Entity` is too high-level for this kind of API, and we already have a some kind of a similar concept encapsulated in the `forms` api.

Reply:
> we should probably be dealing with tables, rows, fields, constraints rather than entities

Happy to talk about naming of resources. I'm not set on anything, but I thought an abstraction over the specific database technology used would be good.

> we already have a some kind of a similar concept encapsulated in the forms api

What exactly are you referring to here?

Reply:
Also I think these conversations are better for the confluence docs. This pr is more about the go api structure. I'm just using core-db as a reason to do it.

Reply:
Yeah sure, I think its really useful to have a concrete example such as this PR to have better idea of what are the general directions & to iterate on !

Reply:
What I mean by that is concerning the conversation we had about exposing low level concepts around `tables`, `records`, etc, its upcoming usefulness in terms of implementing offline, and that the `forms` api is basically syntactic sugar that makes it easy to create those complicated, low-level objects. The forms-api is a higher level concept that depends on the db-api. Wheras the db-api is basically the lowest level thing in core, and the most important one probably.

What I foresee is that a subset of this `core-db-api` will basically become our reconciliation endpoint for offline use. 

There are 2 apis I see in this microservice
1. Management api to create tables, constraints, stuff like that
2. Data api, to push data to those tables

The write part of (1) - Management API, should be exclusively for online use
The read part of (1) - Management Api, closely resemble what we need in terms of forms discovery for offline
The (2) - Data Api, closely resemble what we need in terms of apis for data reconciliation for offline

And whatever we come up with this Data Api will probably be what we need to implement on the client side as well, when we implement cross-device reconciliation. So this Data Api should be written in a way that it is agnostic to the fact that it is a server or a client or whatever else. 

We don't want the mobile devices or website to have a different api whether its talking to a server or to another client in p2p mode. We probably want this api to be exactly the same. So imho the most sensible part of this whole PR is exactly in the design of this data api.

Reply:
Yeah I think I pretty much agree with you. It's the interface of the `CoreDBClient` that should stay the same between offline and online. Seen as the client basically just maps to api endpoints then the api interface should also be fairly consistent.

Reply:
> tables, rows, fields, constraints rather than entities

When you talk about the naming here, do you mean we keep the same data structures but rename them or are you talking about changing the data structures?

Reply:
I think it makes sense to have a DB service, but the entities within this service should be form the DB domain, such as `tables` `fields` `constraints` etc. rather than `entities`

Reply:
Do you mean that for the api interface rather than `GET /entity`, you would have:
```
GET /table
GET /field
GET /constraint
```
or would the `table` entity still encompass the `fields` and `constraints`?